django/shortcuts.py

```diff
@@ -24,37 +25,64 @@
                        context_instance=_context_instance_undefined,
                        content_type=None, status=None, dirs=_dirs_undefined,
                        dictionary=_dictionary_undefined, using=None):
     """
-    Returns a HttpResponse whose content is filled with the result of calling
-    django.template.loader.render_to_string() with the passed arguments.
+    Returns a response_class whose content is filled with the result of calling
+    func() with the passed arguments.
     """
     if (context_instance is _context_instance_undefined
             and dirs is _dirs_undefined
             and dictionary is _dictionary_undefined):
         # No deprecated arguments were passed - use the new code path
-        content = loader.render_to_string(template_name, context, using=using)
+        content = func(template_name, context, using=using)
 
     else:
         # Some deprecated arguments were passed - use the legacy code path
-        content = loader.render_to_string(
+        content = func(
             template_name, context, context_instance, dirs, dictionary,
             using=using)
 
-    return HttpResponse(content, content_type, status)
+    return response_class(content, content_type, status)
 
 
-def render(request, template_name, context=None,
-           context_instance=_context_instance_undefined,
-           content_type=None, status=None, current_app=_current_app_undefined,
-           dirs=_dirs_undefined, dictionary=_dictionary_undefined,
-           using=None):
+def render_to_response(template_name, context=None,
+                       context_instance=_context_instance_undefined,
+                       content_type=None, status=None, dirs=_dirs_undefined,
+                       dictionary=_dictionary_undefined, using=None):
     """
     Returns a HttpResponse whose content is filled with the result of calling
     django.template.loader.render_to_string() with the passed arguments.
+    """
+    return http_response_with(loader.render_to_string, HttpResponse, template_name, context,
+                              context_instance, content_type, status, dirs,
+                              dictionary, using)
+
+
+def stream_to_response(template_name, context=None,
+                       context_instance=_context_instance_undefined,
+                       content_type=None, status=None, dirs=_dirs_undefined,
+                       dictionary=_dictionary_undefined, using=None):
+    """
+    Returns a StreamingHttpResponse whose content is filled with the result of calling
+    django.template.loader.stream() with the passed arguments.
+    """
+    return http_response_with(loader.stream, StreamingHttpResponse, template_name, context,
+                              context_instance, content_type, status, dirs,
+                              dictionary, using)
+
+
+def request_context_response_with(func, response_class, request, template_name, context=None,
```

Comment:
Lately, we've been preferring hanging indent so everything isn't stuck on the right side/indented non-multiple-of-4:
```
def request_context_response_with(
        func, response_class, ...)
```
We could also update the docstrings to "Return ..." while they are being touched.